### ISSUE TITLE
Cannot add a photo to principal or advisor profiles

### ISSUE THREAD

**Issue by dspCarlson** [NCI-Agency/anet#3528]
**Describe the bug**
When editing a principal or advisor profile, I can edit the avatar and select an avatar.  However, when I click the blue "Save Person" button, a pink error message is displayed at the top of the page: TypeError: Network error: NetworkError when attempting to fetch resource.

**To Reproduce**

Condition 1: A principal or advisor profile is displayed.

1. On the profile/person page, select the blue "Edit" button at the top-right of the page.  Result: the profile page will be in edit mode and data fields can be modified
2. Select the gray "Edit avatar" button on in the top left of the page.  Result: the "Edit avatar" pop-up window will be displayed
3. Select the black text-white background "Choose a file" icon in the middle of the "Edit avatar" pop-up window.  Result: A Windows "File Upload" window is displayed
4. From the Windows "File Upload" window, select a profile photo in JPG file format.  Result: the photo will be displayed as an oval in the "Edit avatar" pop-up window
5. Select the white "Save" button on the bottom-right of the "Edit avatar" pop-up window.  Result: the "Edit avatar" pop-up window is no longer displayed AND the photo is now displayed in the profile's avatar
6. Select the blue "Save Person" button in the top-right of the profile page.  Result: A pink error message is displayed at the top of the page: TypeError: Network error: NetworkError when attempting to fetch resource.

**Expected behavior**
When selecting the blue "Save Person" button, the profile should be saved and the photo should be displayed in the avatar field on the profile page.

**Screenshots**
None

**Environment (please complete the following information):**
 - Browser: Firefox
 - ANET production version 2.1.44; not sure when this bug was released

**Additional context**
All other profile data can be saved, e.g. Name, Email, Phone, Rank, Gender, Nationality Code, End of Tour, Biography, etc.

**Comment by dspCarlson** [comment]
This bug was present in ANET Windows production version 2.1.44.  However, the bug is not present in ANET Windows staging 2.1.46.  Since version 2.1.46 will be released to production on Tuesday evening, 6 April 2021, this bug can wait four more days.  I am closing this issue.

**Comment by dspCarlson** [comment]
Upon further investigation, this bug is sometimes present and sometimes not present.  For example, I can try to load a photo to a profile and receive the error message.  Then two hours later, I can upload photos again to profiles.

**Comment by dspCarlson** [comment]
@gjvoosten wrote in Slack: The first one looks like erroneous entries in the dictionary for the showPageOrderedFields ; it should list only fields that are defined for that object. The field names I see in that error above are the ones we use in development, so if they are in the production dictionary, that sure isn't right.

@midmarch wrote in Slack:  I think I fixed the configuration issue on the staging, anet-dictionary.yml  excluded sample fields on line ~547.

@gjvoosten wrote in Slack: The 2nd error is in draft-js, so that is unrelated to the avatar (draft-js is only used in rich-text fields).

@midmarch wrote in Slack: There is a field called code which is not visible in the advisor person profile.

@gjvoosten wrote in Slack: 
Ha! Found it! If you don't have a label defined for code: under:
fields:
  person:
    firstName: First name
    lastName: Last name
    …
    code: ID card number
then it won't display.

@gjvoosten wrote in Slack: And then make sure it is also under fields: advisor: person: showPageOrderedFields: …
